### PR TITLE
Connect Google Cloud SQL to Go API

### LINKED ISSUE #1: Need to Connect App Engine to Cloud SQL

### LINKED ISSUE #3: Need to create models for the database
We need Users and Comments that are related to the users

### PR DESCRIPTION
**Pull request by swoldemi**
This PR closes #1 and closes #3

### CODE REVIEW

File: pkg/database/db.go [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+	}
+	if connectionName == "" {
+		panic(errors.New("SQL_CONNECTION_NAME var is empty"))
+	}
+	if databaseName == "" {
+		panic(errors.New("SQL_DATABASE var is empty"))
+	}
+	dataSourceName := fmt.Sprintf("%s:%s@cloudsql(%s)/%s?charset=utf8&parseTime=True&loc=Local", username, password, connectionName, databaseName)
+
+	db, err := gorm.Open("mysql", dataSourceName)
```
Comment (ynigoreyes) [inline pkg/database/db.go:35]: Does this mean that the development db is also Cloud SQL? Should we not use a local DB?

**Review by ynigoreyes — CHANGES_REQUESTED** [review]
Can you please update the README to make sure we keep track of what we did to get Cloud SQL running?

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,6 @@
 .next
 out
 node_modules
-**gin-bin*
\ No newline at end of file
+**gin-bin*
+.env
+resume-engine-cloud-sql.json
```
Comment (ynigoreyes) [inline .gitignore:6]: Can you send me this? Or how do I get it on my own?

File: pkg/database/db.go [hunk 3]
```diff
@@ -1,15 +1,40 @@
...
+	"github.com/jinzhu/gorm"
+	"log"
+	"os"
 )
 
-// Init initializes an instance of the database
-func Init() (*sql.DB, error) {
-	db, err := sql.Open("mysql", "root@cloudsql(project-id:instance-name)/dbname")
+// MustInit initializes an instance of the database
+func MustInit() *gorm.DB {
```
Comment (ynigoreyes) [inline pkg/database/db.go]: Yeah i remember Josh saying that func Init shouldn't return anything and I was about to ask you lol

**Comment by ynigoreyes** [comment]
Also, since you are doing the Environments too, take into account that you can do development deployments using `app.yaml` and you can declare production environment variables in app.yaml. If we were able to ignore changes to the app.yaml and then populate it with production environment variables, that would be ideal. I tried this right before you started working on it haha

File: pkg/database/db.go (hunk 1 above)
Comment (swoldemi) [reply pkg/database/db.go:35]: Yes, but can create a dev db in the cloud instance. The instance houses several databases. We would just need to change an environment variable. `main` would be prod, and `dev` could be used for development.

File: .gitignore (hunk 2 above)
Comment (swoldemi) [reply .gitignore:6]: Sent. I think you can redownload it from IAM > Service Accounts too.

File: pkg/database/db.go (hunk 3 above)
Comment (swoldemi) [reply pkg/database/db.go]: Changed in 64452d2c75d13566f91b0608d34ae9873426a337

**Review by ynigoreyes — APPROVED** [review]

**Pull request merged**